operators/src/processing/temporal_raster_aggregation/mean_aggregation_subquery.rs

```diff
@@ -0,0 +1,227 @@
+
+                for ((av, ac), iv) in accu_grid
+                    .data
+                    .iter_mut()
+                    .zip(self.count_grid.data.iter_mut())
+                    .zip(in_tile_grid.data.iter())
+                {
+                    if in_tile_grid.is_no_data(*iv) {
+                        *ac = 0;
+                    } else {
+                        let ivf: f64 = iv.as_();
+                        *av = ivf;
+                        *ac = 1;
+                    }
+                }
+
+                self.value_grid = accu_grid.into();
+            }
+
+            GridOrEmpty::Grid(accu_grid) => {
+                for ((av, ac), iv) in accu_grid
+                    .data
+                    .iter_mut()
+                    .zip(self.count_grid.data.iter_mut())
+                    .zip(in_tile_grid.data.iter())
+                {
+                    if in_tile_grid.is_no_data(*iv) {
+                        // The input pixel value is nodata
+                        if !self.ignore_no_data {
+                            // once nodata always nodata
+                            *ac = 0;
+                        }
+                    } else {
+                        let ivf: f64 = iv.as_();
+                        if self.ignore_no_data || *ac > 0 {
+                            // we either ignore nodata, then we add all non-nodata pixels or the count is > 0, so not nodata
+                            // *av += ivf;
+                            *ac += 1;
+                            let delta = ivf - *av;
+                            *av += delta / (*ac as f64);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl<T> FoldTileAccu for TemporalMeanTileAccu<T>
+where
+    T: Pixel,
+{
+    type RasterType = T;
+
+    fn tile_ref(&self) -> &RasterTile2D<Self::RasterType> {
+        unimplemented!();
+    }
+
+    fn into_tile(self) -> RasterTile2D<Self::RasterType> {
+        let TemporalMeanTileAccu {
+            time,
+            tile_position,
+            global_geo_transform,
+            value_grid,
+            count_grid,
+            ignore_no_data: _,
+            out_no_data_value,
+        } = self;
+
+        let value_grid = match value_grid {
+            GridOrEmpty::Grid(g) => g,
+            GridOrEmpty::Empty(_) => {
+                return RasterTile2D::new(
+                    time,
+                    tile_position,
+                    global_geo_transform,
+                    EmptyGrid2D::new(value_grid.grid_shape(), out_no_data_value).into(),
+                )
+            }
+        };
+
+        let res: Vec<T> = value_grid
+            .data
+            .into_iter()
+            .zip(count_grid.data.into_iter())
+            .map(|(v, c)| {
+                if c == 0 {
+                    out_no_data_value
+                } else {
+                    T::from_(v) // lets hope that this works..
```

Comment:
Why shouldn't it work? You created tests for it.

Or do you mean we have f64 and now its u8?

Reply:
yes i was thinking about f64 to u8